maven/lib/dependabot/maven/file_parser/property_value_finder.rb

```diff
@@ -69,6 +73,16 @@ def property_details(property_name:, callsite_pom:)
 
         attr_reader :dependency_files
 
+        def extract_value_from_expression(expression_value:, property_name:, callsite_pom:)
+          # and the expression is pointing to self then raise the error
+          if expression_value.eql?("${#{property_name}}")
+            raise Dependabot::DependencyFileNotParseable, "[ERROR] Resolving expression: '#{expression_value}'"
```

Comment:
We don't need to preface the message with `[ERROR]`, the logging code will already make sure to format it in a way where that's clear, also, at this point we know that it is a recursive cycle I think so maybe we can say that, something like:

```suggestion
            raise Dependabot::DependencyFileNotParseable, "Error trying to resolve recursive expression '#{expression_value}'."
```